### PR TITLE
Adding a CardResult to the response provided by the CardClient

### PR DESCRIPTION
**Pull request by jonathajones**
These changes are steps towards aligning on the result types vended to merchant: https://engineering.paypalcorp.com/jira/browse/DTNOR-203

### Reason for changes
We replaced the `OrderData` response data with a card specific `CardResult` type. This will help us provide more module specific information back to the merchant, since the data we want to vend might be different for the Card module vs the PayPal module. We considered having these results backed by a shared protocol in core, but didn't think it added much value. 

### Summary of changes
- Removing OrderData from Core, replacing it with Card specific `CardResult` type. 

- Updated the expected `ResponseType` conformance from `Codable` to `Decodable`

### Checklist

~ [ ] Added a changelog entry ~

### Authors
- @scannillo @jonathajones

### CODE REVIEW

File: Sources/Card/ConfirmPaymentSourceResponse.swift [hunk 1]
```diff
@@ -1,6 +1,21 @@
...
+import PaymentsCore
+#endif
 
-struct ConfirmPaymentSourceResponse: Codable {
-    var id: String
-    var status: String
+struct ConfirmPaymentSourceResponse: Decodable {
+    let id: String
+    let status: OrderStatus
+
```
Comment (sshropshire) [inline Sources/Card/ConfirmPaymentSourceResponse.swift]:
```suggestion
```

File: Sources/Card/Models/CardResult.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import Foundation
+#if canImport(PaymentsCore)
+import PaymentsCore
+#endif
+
+/// The result of a card payment flow.
+public struct CardResult {
+
+    /// The order ID associated with your transaction
```
Comment (sshropshire) [inline Sources/Card/Models/CardResult.swift]:
```suggestion
    /// The order ID associated with the transaction
```

**Review by sshropshire — APPROVED** [review]
🙌

File: Sources/PaymentsCore/Enums/OrderStatus.swift [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import Foundation
 
 /// The status of an order
-public enum OrderStatus: String {
+public enum OrderStatus: String, Codable {
```
Comment (minhthenguyen) [inline Sources/PaymentsCore/Enums/OrderStatus.swift:4]: ope I didn't notice this wasn't `Codable` before 😮 
I wonder how could it work before, are String enums automatically codable without having to conform somehow? :-ss

File: Sources/Card/Models/CardResult.swift [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import Foundation
+#if canImport(PaymentsCore)
+import PaymentsCore
+#endif
+
+/// The result of a card payment flow.
+public struct CardResult {
```
Comment (minhthenguyen) [inline Sources/Card/Models/CardResult.swift:7]: I just realized if we pass header field `Prefer: return=representation`, we'll be able to get `intent` in the response.
Do you think we should add it to the request, then add `intent` here so merchant won't have to keep track of `intent` by themselves (plus make it more consistent with paypal where we have `ApprovalData.intent`) ? :smile:

**Review by minhthenguyen — APPROVED** [review]
I just have two questions, but everything looks good to me :smile:

File: Sources/Card/Models/CardResult.swift (hunk 4 above)
Comment (scannillo) [reply Sources/Card/Models/CardResult.swift:7]: This is a good callout! I think we should check in w/ Android folks on this, see if it's something we want to add. We can add it in additional PR if so. How does that sound to you?

Comment (minhthenguyen) [reply Sources/Card/Models/CardResult.swift:7]: ah yep that makes sense!!!! :smile:

**Pull request merged**